src/Sylius/Bundle/CoreBundle/Controller/OrderController.php

```diff
@@ -232,6 +121,39 @@ public function saveAction(Request $request)
      *
      * @return Response
      */
+    public function thankYouAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $orderId = $request->getSession()->get('sylius_order_id', null);
+
+        if (null === $orderId) {
+            return $this->redirectHandler->redirectToRoute(
+                $configuration,
+                $configuration->getParameters()->get('after_failure[route]', 'sylius_shop_cart_summary', true),
```

Comment:
That route is from `Shop`, so `Core` is now coupled with it... 

Btw. I would rather redirect on homepage.